show-device-location-using-indoor-positioning/src/main/java/com/esri/arcgismaps/sample/showdevicelocationusingindoorpositioning/MainActivity.kt

```diff
@@ -0,0 +1,361 @@
+     */
+    private fun setUpMap() {
+        // load the portal and create a map from the portal item
+        val portalItem = PortalItem(
+            Portal("https://www.arcgis.com/"),
+            "8fa941613b4b4b2b8a34ad4cdc3e4bba"
+        )
+        mapView.map = ArcGISMap(portalItem)
+        val map = mapView.map
+        lifecycleScope.launch {
+            map?.load()?.onSuccess {
+                val featureTables = map.tables
+                // check if the portalItem contains featureTables
+                if (featureTables.isNotEmpty()) {
+                    setUpLoadTables(featureTables)
+                } else {
+                    showError("Map does not contain feature tables")
+                }
+            }?.onFailure {
+                // if map load failed, show the error
+                showError("Error Loading Map")
+            }
+        }
+    }
+
+    /**
+     * Load each feature table and setup the IndoorsLocationDataSource for each table loaded
+     */
+    private fun setUpLoadTables(featureTables: MutableList<FeatureTable>) {
+        lifecycleScope.launch {
+            featureTables.forEach { featureTable ->
+                // load each FeatureTable
+                featureTable.load().onSuccess {
+                    setupIndoorsLocationDataSource(featureTable)
+                }.onFailure {
+                    showError("Error loading FeatureTable: ${it.message}")
+                }
+            }
+        }
+    }
+
+    /**
+     * Sets up the [mIndoorsLocationDataSource] using the IPS_Positioning table
+     */
+    private fun setupIndoorsLocationDataSource(featureTable: FeatureTable) {
+        // positioningTable needs to be present
+        if (featureTable.tableName == "IPS_Positioning") {
+            val positioningFeatureTable = featureTable as ServiceFeatureTable
+            // when multiple entries are available, IndoorsLocationDataSource constructor function
+            // looks up the entry with the most recent date and takes this positioning data
+            // set up queryParameters to grab one result.
+            val dateCreatedFieldName =
+                getDateCreatedFieldName(positioningFeatureTable.fields)
+            if (dateCreatedFieldName == null) {
+                showError("The service table does not contain \"DateCreated\" fields.")
+                return
+            }
+            val queryParameters = QueryParameters().apply {
+                maxFeatures = 1
+                whereClause = "1 = 1"
+                // find and sort out the orderByFields by most recent first
+                orderByFields.add(
+                    OrderBy(
+                        dateCreatedFieldName,
+                        sortOrder = SortOrder.Descending
+                    )
+                )
+            }
+            lifecycleScope.launch {
+                positioningFeatureTable?.queryFeatures(queryParameters)
+                    ?.onSuccess { queryResults ->
```

Comment:
```suggestion
                positioningFeatureTable.queryFeatures(queryParameters)
                    .onSuccess { queryResults ->
```

Looks like null checks are not needed along this chain.